### ISSUE TITLE
Unable to set Memory Limit of serivce to value of 2GB or greater

### ISSUE THREAD

**Issue by 1Four1Two** [pulumi/pulumi-docker#1185]
### Describe what happened

When trying to create a service with a memory limit of 2GB or higher it will fail. This is because the value must be represented in bytes and the input value is an int. max value being less that the required value

### Sample program

using Pulumi;
using Pulumi.Docker;

class MyStack : Stack
{
    public MyStack()
    {
        var service = new Service("my-docker-service", new ServiceArgs
        {
            Name = "my-service",
            TaskSpec = new ServiceTaskSpecArgs
            {
                ContainerSpec = new ServiceTaskSpecContainerSpecArgs
                {
                    Image = "nginx:latest", // Replace with your desired image
                    Resources = new ServiceTaskSpecContainerSpecResourcesArgs
                    {
                        Limits = new ServiceTaskSpecContainerSpecResourcesLimitsArgs
                        {
                            MemoryBytes = 2 * 1024 * 1024 * 1024 // 2GB memory limit
                        }
                    }
                }
            },
            Mode = new ServiceModeArgs
            {
                Replicated = new ServiceModeReplicatedArgs
                {
                    Replicas = 1 // Number of replicas
                }
            },
            EndpointSpec = new ServiceEndpointSpecArgs
            {
                Ports = 
                {
                    new ServiceEndpointSpecPortsArgs
                    {
                        Protocol = "tcp",
                        TargetPort = 80,
                        PublishedPort = 8080
                    }
                }
            }
        });

        // Export the service ID
        this.ServiceId = service.Id;
    }

    [Output]
    public Output<string> ServiceId { get; set; }
}

class Program
{
    static Task<int> Main() => Deployment.RunAsync<MyStack>();
}

### Log output

rror response from daemon: rpc error: code = InvalidArgument desc = invalid memory value -2147483648: Must be at least 4MiB

### Output of `pulumi about`

CLI          
Version      3.124.0
Go Version   go1.22.5
Go Compiler  gc

Plugins
KIND      NAME     VERSION
resource  aws      6.44.0
resource  command  0.11.1
resource  docker   4.5.4
language  dotnet   unknown

Host     
OS       ubuntu
Version  22.04
Arch     x86_64

This project is written in dotnet: executable='/usr/bin/dotnet' version='7.0.119'

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by blampe** [comment]
I don't think we can expose this as a long without https://github.com/pulumi/pulumi/issues/14242 first.

Alternatively it would be reasonable to accept a string (e.g. "4GB").

Please give this a 👍 if you'd like to see it prioritized!
